packages/nextjs/components/Modal.tsx

```diff
@@ -0,0 +1,44 @@
+/**
+ * Modal component: A reusable modal component.
+ */
+type ModalProps = {
+  trigger: JSX.Element;
+  modalTitle: string;
```

Comment:
We need to use that prop for `htmlsFor` and `id` attributes but not for title. So I think `name` or `id' is better.

Title should be in different place, because it can be empty or long with spaces and can break labels/ids.